### PR TITLE
Added support for fetching what teams are part of this team.

### PR DESCRIPTION
**Pull request by alexanderkjall**
# Description 
The call is to this endpoint https://developer.github.com/v3/teams/#list-child-teams-legacy

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [x] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [x] Fill in the "Description" above. 
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by alexanderkjall** [comment]
I tried following the setup guide for writing tests, but without success, I got:

```
org.kohsuke.github.GHFileNotFoundException: http://localhost:35361/orgs/hub4j-test-org/teams/dummy-team {"message":"Not Found","documentation_url":"https://developer.github.com/v3/teams/#get-team-by-name"}

```

And didn't really understand why that happened.

**Comment by bitwiseman** [comment]
@alexanderkjall  
Please add a test. I can then help you refine it.

**Comment by bitwiseman** [comment]
Also, please describe the steps you followed from the guide and the result from each step.

**Comment by alexanderkjall** [comment]
The problem that I had was due to not being a part of the test organization, once I was added it worked without problem.

File: src/main/java/org/kohsuke/github/GHTeam.java [hunk 1]
```diff
@@ -174,6 +174,21 @@ public GHDiscussion getDiscussion(long discussionNumber) throws IOException {
...
+    /**
+     * Retrieves the teams that are members of this team.
+     *
+     * @param organization
+     *            the organization that owns this team
+     * @return the paged iterable
+     * @throws IOException
+     *             the io exception
+     */
+    public PagedIterable<GHTeam> listChildTeams(GHOrganization organization) throws IOException {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHTeam.java:186]: The team already knows what organization it is part of, doesn't it?

Comment (alexanderkjall) [reply src/main/java/org/kohsuke/github/GHTeam.java:186]: I would have guessed so, but I got an Exception when I tried to use the wrapUp method that didn't take an organization argument.

Is there a better way to solve this?

Comment (alexanderkjall) [reply src/main/java/org/kohsuke/github/GHTeam.java]: I now see that the organization is stored as a member of the Team.

**Pull request merged**